mii/config.py

```diff
@@ -3,93 +3,147 @@
 
 # DeepSpeed Team
 import torch
-from typing import Union, List
-from enum import Enum
-from pydantic import BaseModel, validator, root_validator
-
-from deepspeed.launcher.runner import DLTS_HOSTFILE
-
-
-class DtypeEnum(Enum):
-    # The torch dtype must always be the first value (so we return torch.dtype)
-    fp16 = torch.float16, "torch.float16", "fp16", "float16", "half"
-    bf16 = torch.bfloat16, "torch.bfloat16", "bf16", "bfloat16"
-    fp32 = torch.float32, "torch.float32", "fp32", "float32", "float"
-    int8 = torch.int8, "torch.int8", "int8"
-
-    # Copied from https://stackoverflow.com/a/43210118
-    # Allows us to use multiple values for each Enum index and returns first
-    # listed value when Enum is called
-    def __new__(cls, *values):
-        obj = object.__new__(cls)
-        # first value is canonical value
-        obj._value_ = values[0]
-        for other_value in values[1:]:
-            cls._value2member_map_[other_value] = obj
-        obj._all_values = values
-        return obj
-
-    def __repr__(self):
-        return "<%s.%s: %s>" % (
-            self.__class__.__name__,
-            self._name_,
-            ", ".join([repr(v) for v in self._all_values]),
-        )
+import os
+import string
+from typing import List, Optional, Dict, Any
+from pydantic import validator, root_validator
+import mii
+from mii.constants import DeploymentType, TaskType, MII_MODEL_PATH_DEFAULT
 
+from deepspeed.runtime.config_utils import DeepSpeedConfigModel
+from deepspeed.inference.config import DtypeEnum
+from deepspeed.launcher.runner import DLTS_HOSTFILE, fetch_hostfile
 
-class MIIConfig(BaseModel):
-    tensor_parallel: int = 1
-    port_number: int = 50050
-    dtype: DtypeEnum = torch.float32
-    meta_tensor: bool = False
+
+class ReplicaConfig(DeepSpeedConfigModel):
+    hostname: str = ""
+    tensor_parallel_ports: List[int] = []
+    torch_dist_port: int = None
+    gpu_indices: List[int] = []
+
+
+class DeploymentConfig(DeepSpeedConfigModel):
```

Comment:
Is this the organization of config params we want? Is it worth having model/inferences configs be sub-configs of the deployment config?